devicetypes/smartthings/fibaro-rgbw-controller.src/fibaro-rgbw-controller.groovy

```diff
@@ -600,271 +271,38 @@ def refresh() {
-        delayBetween(cmds, 1500)
-    }
-}
-
-def test() {
-	//def value = [:]
-    //value = [hue: 0, saturation: 100, level: 5]
-    //value = [red: 255, green: 0, blue: 255, level: 60]
-	//setColor(value)
-
-    def cmd = []
-
-    if ( !state.cnt ) {
-    	state.cnt = 6
-    } else {
-    	state.cnt = state.cnt + 1
-    }
-
-    if ( state.cnt > 10 )
-    	state.cnt = 6
-
-    // run programmed light show
-	cmd << zwave.configurationV1.configurationSet(configurationValue: [state.cnt], parameterNumber: 72, size: 1).format()
-    cmd << zwave.configurationV1.configurationGet(parameterNumber: 72).format()
-
-    delayBetween(cmd, 500)
-
-}
-
-def colorNameToRgb(color) {
-
-	final colors = [
-        [name:"Soft White",	r: 255, g: 241, b: 224	],
-        [name:"Daylight", 	r: 255, g: 255, b: 251	],
-        [name:"Warm White", r: 255, g: 244, b: 229	],
-
-        [name:"Red", 		r: 255, g: 0,	b: 0	],
-		[name:"Green", 		r: 0, 	g: 255,	b: 0	],
-        [name:"Blue", 		r: 0, 	g: 0,	b: 255	],
-
-		[name:"Cyan", 		r: 0, 	g: 255,	b: 255	],
-        [name:"Magenta", 	r: 255, g: 0,	b: 33	],
-        [name:"Orange", 	r: 255, g: 102, b: 0	],
-
-        [name:"Purple", 	r: 170, g: 0,	b: 255	],
-		[name:"Yellow", 	r: 255, g: 255, b: 0	],
-        [name:"White", 		r: 255, g: 255, b: 255	]
-	]
-
-    def colorData = [:]
-    colorData = colors.find { it.name == color }
-
-    colorData
-}
+		cmds << zwave.configurationV1.configurationSet(configurationValue: pValue, parameterNumber: pNumber, size: pSize)
 
-private hex(value, width=2) {
-	def s = new BigInteger(Math.round(value).toString()).toString(16)
-	while (s.size() < width) {
-		s = "0" + s
+		cmds << zwave.configurationV1.configurationGet(parameterNumber: pNumber)
+		commands(cmds, 1500)
 	}
-	s
 }
 
-def hexToRgb(colorHex) {
-	def rrInt = Integer.parseInt(colorHex.substring(1,3),16)
-    def ggInt = Integer.parseInt(colorHex.substring(3,5),16)
-    def bbInt = Integer.parseInt(colorHex.substring(5,7),16)
-
-    def colorData = [:]
-    colorData = [r: rrInt, g: ggInt, b: bbInt]
-    colorData
+def rgbToHSV(red, green, blue) {
+	def hex = colorUtil.rgbToHex(red as int, green as int, blue as int)
+	def hsv = colorUtil.hexToHsv(hex)
+	return [hue: hsv[0], saturation: hsv[1], value: hsv[2]]
 }
 
-def rgbToHex(rgb) {
-    def r = hex(rgb.r)
-    def g = hex(rgb.g)
-    def b = hex(rgb.b)
-    def hexColor = "#${r}${g}${b}"
-
-    hexColor
+def huesatToRGB(hue, sat) {
+	def color = colorUtil.hsvToHex(Math.round(hue) as int, Math.round(sat) as int)
+	return colorUtil.hexToRgb(color)
 }
 
-def hslToRGB(float var_h, float var_s, float var_l) {
-	float h = var_h / 100
-    float s = var_s / 100
-    float l = var_l
-
-    def r = 0
-    def g = 0
-    def b = 0
-
-	if (s == 0) {
-    	r = l * 255
-        g = l * 255
-        b = l * 255
+private command(physicalgraph.zwave.Command cmd) {
+	if (zwaveInfo.zw.endsWith("s"))  {
```

Comment:
Worth checking `zwaveInfo?.zw?` anymore? where did we end up on that?

Reply:
I was about to say that it should always exist now, but I remembered this is an old-as-sin dth